relay-replays/src/recording.rs

```diff
@@ -43,104 +41,152 @@ impl From<serde_json::Error> for ParseRecordingError {
+            Some(body) => body,
+        };
+
+        let mut output = header.to_owned();
+        output.push(b'\n');
+        // Data scrubbing usually does not change the size of the output by much. We can preallocate
+        // enough space for the scrubbed output to avoid resizing the output buffer serveral times.
+        // Benchmarks have NOT shown a big difference, however.
+        output.reserve(body.len());
+        self.transcode_replay(body, &mut output)?;
+
+        Ok(output)
     }
 }
 
-/// Parses compressed replay recording payloads and applies data scrubbers.
-///
-/// `limit` controls the maximum size in bytes during decompression. This function returns an `Err`
-/// if decompressed contents exceed the limit.
-pub fn process_recording(bytes: &[u8], limit: usize) -> Result<Vec<u8>, ParseRecordingError> {
-    // Check for null byte condition.
-    if bytes.is_empty() {
-        return Err(ParseRecordingError::Message("no data found"));
+impl Transform for &'_ mut ReplayScrubber<'_> {
+    fn transform_str<'a>(&mut self, v: &'a str) -> Cow<'a, str> {
+        self.transform_string(v.to_owned())
     }
 
-    let mut split = bytes.splitn(2, |b| b == &b'\n');
-    let header = split
-        .next()
-        .ok_or(ParseRecordingError::Message("no headers found"))?;
-
-    let body = match split.next() {
-        Some(b"") | None => return Err(ParseRecordingError::Message("no body found")),
-        Some(body) => body,
-    };
-
-    let mut output = header.to_owned();
-    output.push(b'\n');
-    // Data scrubbing usually does not change the size of the output by much. We can preallocate
-    // enough space for the scrubbed output to avoid resizing the output buffer serveral times.
-    // Benchmarks have NOT shown a big difference, however.
-    output.reserve(body.len());
-    transcode_replay(body, limit, &mut output)?;
-
-    Ok(output)
+    fn transform_string(&mut self, mut value: String) -> Cow<'static, str> {
+        if let Some(ref mut processor) = self.processor1 {
+            if processor
+                .process_string(&mut value, &mut Meta::default(), &STRING_STATE)
+                .is_err()
+            {
+                return Cow::Borrowed("");
+            }
+        }
+
+        if let Some(ref mut processor) = self.processor2 {
+            if processor
+                .process_string(&mut value, &mut Meta::default(), &STRING_STATE)
+                .is_err()
+            {
+                return Cow::Borrowed("");
+            }
+        }
+
+        Cow::Owned(value)
+    }
 }
 
 #[cfg(test)]
 mod tests {
     // End to end test coverage.
 
+    use relay_general::pii::PiiConfig;
+    use relay_general::processor::SelectorSpec;
+
+    use super::ReplayScrubber;
+
+    fn default_pii_config() -> PiiConfig {
+        let mut pii_config = PiiConfig::default();
+        pii_config.applications = [(SelectorSpec::And(vec![]), vec!["@common".to_string()])].into();
+        pii_config
+    }
```

Comment:
To have the tests resemble the default production use case, I would use a configuration derived from `DataScrubbingConfig` here, like for example https://github.com/getsentry/relay/blob/afa6576707a77c45ed61b277f7900a0e2f835ba5/relay-general/src/protocol/replay.rs#L554-L560

Reply:
Thanks, updated. This actually shows us that the filter has changed.

@cmanallen note that email addresses aren't being scrubbed anymore, apparently the default data scrubbers don't do this. No action item imho, this is in line with the other data scrubbers.